### ISSUE TITLE
Edits fail to validate ranges correctly before applying

### ISSUE THREAD

**Issue by HookyQR** [Microsoft/vscode#6281]
- VSCode Version: 1.1
- OS Version: OSX

Affected `beautify` for VS Code. HookyQR/VSCodeBeautify#14
Steps to Reproduce:

Create extension and, with a file open with content:

```javascript
const active = vscode.window.activeTextEditor;
let range = new vscode.Range(0, 0, Number.MAX_VALUE, Number.MAX_VALUE);
active.edit(edit => edit.replace(range,"new"));
```
Previously (v1.0 and lower) this would replace the file content with "new" as the edit call would validate the range by reducing the infinite end to the end of the file. From version 1.1 (including insider) the infinite end is set to zero instead.

`document.validateRange` still performs as expected.

I have changed my extension to call `validateRange` before any edits, but I believe the way the edits worked previously is more logical.

**Comment by alexandrudima** [comment]
@HookyQR Thank you very much for the heads up, we will be including this fix in 1.1.1

**Comment by proutorov** [comment]
test
